File: epi_judge_java/epi/MakingChange.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class MakingChange {

    @EpiTest(testDataFile = "making_change.tsv")
    public static int changeMaking(int cents) {
        int[] coins = {100, 50, 25, 10, 5, 1};
        int ans = 0;
        for (int coin : coins) {
            ans += cents / coin;
            cents %= coin;
        }
        return ans;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "MakingChange.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/making_change.cc
#include "test_framework/generic_test.h"

const int COINS[] = {100, 50, 25, 10, 5, 1};

int ChangeMaking(int cents) {
    int result = 0;
    for (int coin : COINS) {
        result += cents / coin;
        cents %= coin;
    }
    return result;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"cents"};
    return GenericTestMain(args, "making_change.cc", "making_change.tsv",
                           &ChangeMaking, DefaultComparator{}, param_names);
}
